src/instruction.rs

```diff
@@ -213,143 +213,239 @@ impl fmt::Display for MemoryReference {
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub struct Gate {
+    pub name: String,
+    pub parameters: Vec<Expression>,
+    pub qubits: Vec<Qubit>,
+    pub modifiers: Vec<GateModifier>,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct CircuitDefinition {
+    pub name: String,
+    pub parameters: Vec<String>,
+    // These cannot be fixed qubits and thus are not typed as `Qubit`
```

Comment:
Would splitting the `Qubit` type into fixed & variable be useful? Or just more clutter?

Reply:
Since each just wraps a primitive type, I feel like this would just be clutter, but I'm not fixed on that